test/screenshot/tab/index.tsx

```diff
@@ -0,0 +1,134 @@
+import * as React from 'react';
+import Tab, {TabProps} from '../../../packages/tab';
+// https://github.com/material-components/material-components-web-react/issues/513
+// @ts-ignore
+import MaterialIcon from '../../../packages/material-icon/index';
+import '../../../packages/tab-indicator/index.scss';
+import '../../../packages/tab/index.scss';
+import './index.scss';
+
+interface TabContentType {
+  num: number;
+}
+
+const TabContent: (args: TabContentType) => React.ReactElement<any> = ({
+  num, // eslint-disable-line react/prop-types
+}) => (
+  <React.Fragment>
+    <MaterialIcon className='mdc-tab__icon' icon='favorite' />
+    <span className='mdc-tab__text-label'>Tab {num}</span>
+  </React.Fragment>
+);
+
+interface TabsProps {
+  children: React.ReactNode;
+}
+
+function isElementTab(element: any): element is Tab {
+  return element !== null;
+}
+
+const Tabs: (args: TabsProps) => JSX.Element = ({
+  children, // eslint-disable-line react/prop-types
+}) => {
+  return <div className='tabs'>{children}</div>;
+};
+
+interface TabsControllerProps extends TabProps {
+  tabContent: (num: number) => React.ReactElement<any>
+}
+
+interface TabsControllerState {
+  activeIndex: number;
+  previousActiveIndex: number;
+}
+
+class TabsController extends React.Component<TabsControllerProps, TabsControllerState> {
+  tabBoundingRects: {[n: number]: ClientRect} = {};
+
+  state = {activeIndex: 0, previousActiveIndex: 0};
+
+  tabInit = (tabEl: React.Ref<Tab>, index: number) => {
+    if (isElementTab(tabEl)) {
+      this.tabBoundingRects[index] = tabEl.computeIndicatorClientRect();
+    }
+  }
+
+  render() {
+    const {
+      tabContent, // eslint-disable-line
+      ...otherProps
+    } = this.props;
+
+    const {activeIndex, previousActiveIndex} = this.state;
+
+    return (
+      <Tabs>
+        {[1, 2, 3].map((num, index) => (
+          <Tab
+            active={index === activeIndex}
+            key={index}
+            previousIndicatorClientRect={this.tabBoundingRects[previousActiveIndex]}
+            // https://github.com/DefinitelyTyped/DefinitelyTyped/issues/28884
+            // @ts-ignore
+            ref={(el: React.Ref<Tab>) => this.tabInit(el, index)}
```

Comment:
TS Style says you have to use `=> { /*code*/ }` if you don't use the result of a lambda.

Reply:
I think your type is wrong here, and that this should not be `ts-ignore`d.  This does not use the `styled-components` library, which is what the linked issue is concerning.

If you look at the [typings for `Ref`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L65), you'll see that it's either a function that is passed a `T`, or the result of `createRef()`.  You've confused the two here.

`(el: Tab)`  is the correct signature.  You might not even need the `: Tab`, since TypeScript can often infer local types.